### DISCUSSION TITLE
Issues in Location search API

### DISCUSSION THREAD

**Discussion by priyanka-TL** [Sunbird-Lern/Community#35] [Issues] [answered]
@amit-tarento @reshmi-nair

We found some issues in the location search API. Please check and let us know.

**Issue 1**
We were trying to get the child locationIds using the below curl request, here **count** value in the response is showing 670, but we count the length of the response array we got 250 as count. the count is mismatching
```
curl --location --request POST 'https://staging.sunbirded.org/api/data/v1/location/search' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJSbTBGOTlnMnd4SmZKelJwa2YzTnRlaE5CYlJnZXRCaCJ9.CjDN6GufxylpAM5J_6j9fD0wq7S2qr1F6FOzAZtQ6XU' \
--header 'content-Type: application/json' \
--data-raw '{
    "request": {
        "filters": {
            "parentId":[ "8ac1efe9-0415-4313-89ef-884e1c8eee34",
  "98ae45d7-9257-4c14-a16a-9760a442ff28",
  "b5c35cfc-6c1e-4266-94ef-a425c43c7f4e",
  "b617e607-0a5b-45a0-9894-7a325ffa45c7",
  "03f321d5-c398-4315-8d7c-58b92242d7ab",
  "732b83e7-cf4f-401c-a374-db1d45644b3b",
  "aecac7ab-15e4-45c9-ac7b-d716444cd652",
  "87422ed0-d2dd-4672-9d6b-10a4b565dfe3",
  "df1bdc4e-7352-4be2-8e81-f9eb9f252286",
  "2f76dcf5-e43b-4f71-a3f2-c8f19e1fce03",
  "b54a5c6d-98be-4313-af1c-33040b1703aa",
  "a31156d9-2d3f-4eea-84ee-08a2a79cc3db",
  "9346a6ba-0b31-4893-babc-a5be417e8e09" ]
        }
    }
}'
```

**Issue 2**
For the same above curl, 250 block data we are getting. But when we search a district id “8ac1efe9-0415-4313-89ef-884e1c8eee34 ” as **parentId** we are getting 38 results. Ideally, all these 38 results should be present in the 250 count result, but only 10 data we found in that 250, 28 blocks are missing.

cc - @aks30, @VISHNUDAS-tunerlabs

**Comment by amitec006** [comment, marked as answer]
@priyanka-TL This is not an issue, its default behaviour of any search API of lern BB.
By default search API will return you 250 matched documents from ES in response. 
Its better you should use pagination or else if you need complete data in single API call response then pass the limit field in request with whatever doc count you need in response.
